napari/layers/points/points.py

```diff
@@ -1624,8 +1636,17 @@ def remove_selected(self):
                 self.text.remove(index)
             if self._value in self.selected_data:
                 self._value = None
-            self.selected_data = set()
+            else:
+                if self._value is not None:
+                    # update the index of self._value to account for the
+                    # data being removed
+                    indices_removed = np.array(index) < self._value
+                    offset = np.sum(indices_removed)
+                    self._value -= offset
+                    self._value_stored -= offset
+
```

Comment:
Was this a bug getting fixed, or is it more related with 3D specifically than it seems? If yes, how?

Reply:
Yeah, this was a fix for an unrelated bug that I caught while testing. The bug was:

1. Points selected (2D or 3D)
2. A point outside of the selection is highlighted (i.e., the mouse is hovered over it).
3. Delete the points while still hovering over the point
4. Since the index of the highlighted point is not updated, an `IndexError` is raised if that point had an index value greater than some of those that were deleted.